<a href="https://colab.research.google.com/github/Ishikaaa/ML-Data-Scientist-job-market-in-the-US/blob/master/Data_Scientist_job_market_in_the_US.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

***About The Data - ***

alldata.csv -This data explore the job market around the U.S.. It aggregates all information and cleans the job description 
For those who are actively looking for data scientist jobs in the U.S., the best news this month is the LinkedIn Workforce Report August 2018. According to the report, there is a shortage of 151,717 people with data science skills, with particularly acute shortages in New York City, San Francisco Bay Area and Los Angeles.

To help job hunters (including me) to better understand the job market, I scraped Indeed website and collected information of 7,000 data scientist jobs around the U.S. on August 3rd. The information that I collected are: Company Name, Position Name, Location, Job Description, and Number of Reviews of the Company..

***Aim of The Project - ***

It helps the candidate to find job in the field of Data Scientist in U.S..


# ***1. Importing Libraries - ***

In [0]:
import pandas as pd 
import numpy as np 
import re 
import seaborn as sns 
from sklearn.preprocessing import LabelEncoder,OneHotEncoder 
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn import metrics 

In [0]:
dataset=pd.read_csv("alldata.csv")
dataset.head()

position  ...            location
0                               Development Director  ...  Atlanta, GA 30301 
1  An Ostentatiously-Excitable Principal Research...  ...         Atlanta, GA
2                                     Data Scientist  ...         Atlanta, GA
3                                       Data Analyst  ...  Atlanta, GA 30303 
4  Assistant Professor -TT - Signal Processing & ...  ...         Atlanta, GA

[5 rows x 5 columns]

In [0]:
dataset.shape 

(6964, 5)

In [0]:
dataset.dtypes

position        object
company         object
description     object
reviews        float64
location        object
dtype: object

# ***Handling Missing Data - ***

In [0]:
dataset.isnull().sum()

position         11
company          11
description      11
reviews        1638
location         11
dtype: int64

In [0]:
# removing that rows in which all colunms are null 
dataset = dataset.dropna(how='all',axis=0)
dataset.shape

(6953, 5)

In [0]:
dataset.isnull().sum()

position          0
company           0
description       0
reviews        1627
location          0
dtype: int64

In [0]:
# remove that rows which do not have any review 
dataset['reviews']=dataset['reviews'].fillna(0)
dataset.shape 

(6953, 5)

In [0]:
dataset.isnull().sum()

position       0
company        0
description    0
reviews        0
location       0
dtype: int64

In [0]:
# changing datatype of "reviews" column 
dataset['reviews']=dataset['reviews'].astype(int)
dataset.dtypes

position       object
company        object
description    object
reviews         int64
location       object
dtype: object

# ***EDA - ***

In [0]:
dataset['position'].value_counts()

Data Scientist                                                         351
Senior Data Scientist                                                   96
Research Analyst                                                        64
Data Engineer                                                           60
Machine Learning Engineer                                               56
Lead Data Scientist                                                     31
Research Scientist                                                      30
Sr. Data Scientist                                                      28
Principal Data Scientist                                                24
Software Engineer                                                       22
Data Analyst                                                            22
Quantitative Analyst                                                    21
Scientist                                                               17
Product Manager          

***Note - ***


1.   Maximum job offers are of position "Data Scientist".



In [0]:
dataset['company'].value_counts()

Amazon.com                                                 358
Ball Aerospace                                             187
Microsoft                                                  137
Google                                                     134
NYU Langone Health                                          77
Fred Hutchinson Cancer Research Center                      70
KPMG                                                        66
Lab126                                                      50
Facebook                                                    49
Broad Institute                                             49
Walmart eCommerce                                           45
McKinsey & Company                                          41
Takeda Pharmaceutical                                       39
Harnham                                                     39
University of Washington                                    39
Oath Inc                                               

***Note - ***

1.   Maximum offers of jobs are given "Amazon.com" company. 




In [0]:
dataset['reviews'].value_counts()

0         1627
25902      224
108        193
2          191
5          138
4          131
4618       118
2474       106
3          101
6           94
25895       89
7           78
705         76
91          71
8           66
15          65
838         62
32          55
11          54
366         49
14          48
141         46
12          45
70          42
26          42
10          41
385         41
52          39
530         39
17          36
          ... 
189          1
207          1
221          1
233          1
897          1
893          1
2936         1
63           1
2828         1
75           1
653          1
645          1
148085       1
99           1
8282         1
477          1
441          1
413          1
397          1
115          1
119          1
127          1
131          1
325          1
297          1
24869        1
285          1
2210         1
175          1
231          1
Name: reviews, Length: 561, dtype: int64

In [0]:
dataset['description'][0]

"Development Director\nALS Therapy Development Institute has an immediate opening for Development Directors. Reporting directly to the Senior Development Director, the Development Director at ALS TDI is a senior fundraising position working to identifying potential prospects and cultivating solicitation strategies and in closing asks with donors including individuals and corporations by building networks via events, generating awareness of ALS TDI; outreach including attending and speaking at events as well as personally cultivates relationships with patients, prospects and donors. This position will be responsible for generating and managing a portfolio of at least two million to five million dollars per year. This position will be located in Atlanta, GA.\n\nRequirements:\nBachelor's Degree requiredMinimum 6-8 years' experience in fundraising and business developmentSuccessful track recording in fundraising with major donors or scientific sales preferredDemonstrated ability to work in

In [0]:
dataset['location'].value_counts()

Seattle, WA                563
New York, NY               508
Cambridge, MA              487
Boston, MA                 454
San Francisco, CA          425
San Diego, CA              294
Chicago, IL                292
Boulder, CO                233
Washington, DC             230
Sunnyvale, CA              221
Mountain View, CA          199
Los Angeles, CA            187
Austin, TX                 156
Atlanta, GA                147
Redmond, WA 98052          117
Seattle, WA 98109           95
Cambridge, MA 02139         95
New York, NY 10016          83
San Diego, CA 92121         71
Cambridge, MA 02142         64
South San Francisco, CA     53
Chicago, IL 60654           45
San Francisco, CA 94103     40
Chicago, IL 60601           40
Mountain View, CA 94043     39
Redmond, WA                 39
Seattle, WA 98101           34
Seattle, WA 98104           34
Cambridge, MA 02140         33
Boston, MA 02116            31
                          ... 
Seattle, WA 98164            1
San Fran

***Note - ***

1.   Most of the job offers are on the location 'Seattle, WA'.



In [0]:
dataset[dataset['reviews']==dataset['reviews'].max()]

position  ...             location
6728  Staff Software Engineer, Front End Lead  ...  Sunnyvale, CA 94086
6828         Information Security Risk Expert  ...  Sunnyvale, CA 94086
6958         Senior Data Scientist (GEC11902)  ...  Sunnyvale, CA 94086

[3 rows x 5 columns]

In [0]:
dataset[dataset['company']=='Walmart']

position  ...             location
449                        Data Scientist-ISD  ...     Austin, TX 78716
6668                     Staff Data Scientist  ...  San Bruno, CA 94066
6728  Staff Software Engineer, Front End Lead  ...  Sunnyvale, CA 94086
6828         Information Security Risk Expert  ...  Sunnyvale, CA 94086
6958         Senior Data Scientist (GEC11902)  ...  Sunnyvale, CA 94086

[5 rows x 5 columns]

***Note - ***

1.   Company "Walmart" has maximum reviews. And Company "Walmart" offers maximum jobs on location "Sunnyvale, CA 94086".
2.   Maximum offers are at the location "Seattle, WA". But the company with maximum reviews go at "Sunnyvale,CA 94086" location instead of going at location "Seattle, WA". This means that more talented persons are at "Sunnyvale, CA 94086" location. 

In [0]:
dataset[dataset['reviews']==dataset['reviews'].min()]

position  ...             location
0                                  Development Director  ...   Atlanta, GA 30301 
1     An Ostentatiously-Excitable Principal Research...  ...          Atlanta, GA
2                                        Data Scientist  ...          Atlanta, GA
18                                       Data Scientist  ...          Atlanta, GA
19                                       Data Scientist  ...          Atlanta, GA
25                                       Data Scientist  ...          Atlanta, GA
26                 Data Scientist 1 – AP Forensics Team  ...          Atlanta, GA
38                                       Data Scientist  ...          Atlanta, GA
48                        Chief Deep Learning Scientist  ...          Atlanta, GA
53                                  Statistical Analyst  ...          Atlanta, GA
54                                   VP, Data Analytics  ...          Atlanta, GA
55                              Lead Decision Scientist  ...   Atlanta, GA 30313 
65                                   Research Scientist  ...          Atlanta, GA
84                                   Research Scientist  ...          Atlanta, GA
106   Staff Research Scientist VII: Informational In...  ...          Atlanta, GA
114                             Jr Microbiologist - CDC  ...          Atlanta, GA
122               Mid Data Scientist - Machine Learning  ...   Atlanta, GA 30342 
139   Principal Research Scientist I: Informational ...  ...          Atlanta, GA
140                             Market Research Analyst  ...          Atlanta, GA
151                        Senior Scientist, Immunology  ...          Atlanta, GA
153          A Vigorously-Charged Scientific Programmer  ...          Atlanta, GA
154                                   Software Engineer  ...          Atlanta, GA
156                                Director of Software  ...   Atlanta, GA 30332 
157       Certifying Scientist / Quantitation Scientist  ...          Atlanta, GA
161                                      Data Scientist  ...   Atlanta, GA 30326 
164   Windows Programmer / Software Engineer - Inter...  ...          Atlanta, GA
170   Windows Programmer / Software Engineer - Entry...  ...          Atlanta, GA
192   Various Professional, Technical, Administrativ...  ...          Atlanta, GA
196     Business Development Manager (Southeast Region)  ...          Atlanta, GA
199   Implementation Engineer (Machine Learning/Data...  ...   Atlanta, GA 30326 
...                                                 ...  ...                  ...
6789                                     Head of Design  ...  Sunnyvale, CA 94085
6791                 Forward Deployed Solution Engineer  ...        Sunnyvale, CA
6795        Deep Learning Software Development Engineer  ...        Sunnyvale, CA
6800                                 Knowledge Engineer  ...        Sunnyvale, CA
6813  AI Research/Software Development Engineer, Ale...  ...        Sunnyvale, CA
6815                         3D Tools Software Engineer  ...  Sunnyvale, CA 94089
6816       Software Engineer - Tools and Infrastructure  ...        Sunnyvale, CA
6817  23103 Principal Data Scientist - HealthTech / ...  ...        Sunnyvale, CA
6822                                      Manager, TPMs  ...        Sunnyvale, CA
6824                               QA Software Engineer  ...        Sunnyvale, CA
6825                                Sr Speech Scientist  ...        Sunnyvale, CA
6827                      Software Development Engineer  ...        Sunnyvale, CA
6829                                  Software Engineer  ...        Sunnyvale, CA
6841                      Sr. Machine Learning Engineer  ...        Sunnyvale, CA
6846                        Sr. Hardware Audio Engineer  ...        Sunnyvale, CA
6848                       Materials (Quality) Engineer  ...  Sunnyvale, CA 94086
6856  Senior Data Scientist, Algorithms and Machine ...  ...        Sunnyvale, CA
6868                    Senio

# ***Adding new columns - ***

In [0]:
# Adding new column "City"
a=[]
for i in dataset["location"]:
    b=i.split(",")[0]
    a.append(b)
dataset["City"]=a    
dataset.tail()

position  ...       City
6959  Data Developer / Machine Learning Analyst  ...  Sunnyvale
6960                                Scientist I  ...  Sunnyvale
6961                           Intern Scientist  ...  Sunnyvale
6962            Senior Data & Applied Scientist  ...  Sunnyvale
6963    Principal Data Scientist, Deep Learning  ...  Sunnyvale

[5 rows x 6 columns]

In [0]:
dataset["City"].value_counts()

New York                  848
Seattle                   777
Cambridge                 694
Boston                    629
San Francisco             564
Chicago                   471
San Diego                 412
Washington                340
Mountain View             277
Atlanta                   269
Sunnyvale                 256
Los Angeles               254
Boulder                   251
Austin                    213
Redmond                   158
South San Francisco        73
Redwood City               59
Oakland                    35
San Mateo                  30
Manhattan                  30
San Francisco Bay Area     26
Berkeley                   19
Emeryville                 19
Jersey City                17
San Carlos                 16
Queens                     15
Brooklyn                   14
Foster City                14
Menlo Park                 14
Novato                     12
                         ... 
West Orange                 1
Cedar Park                  1
Union City

***Note - ***


1.   "New York" city has maximum job offers. 



In [0]:
# Adding new column "State"
aa=[]
for ii in dataset["location"]:
    bb=ii.split(",")[1]
    aa.append(bb)
for ii in range(len(aa)):
    bb=re.sub(r'[0-9]',"",aa[ii]).strip()
    aa[ii]=bb  
dataset["State"]=aa
dataset.tail()

position  ... State
6959  Data Developer / Machine Learning Analyst  ...    CA
6960                                Scientist I  ...    CA
6961                           Intern Scientist  ...    CA
6962            Senior Data & Applied Scientist  ...    CA
6963    Principal Data Scientist, Deep Learning  ...    CA

[5 rows x 7 columns]

In [0]:
dataset["State"].value_counts()

CA    2152
MA    1323
WA     935
NY     926
IL     471
DC     340
GA     269
CO     251
TX     218
NJ      68
Name: State, dtype: int64

***Note - ***


1.   "CA" state has maximum job offers. 



In [0]:
# adding a new column "Data Scientists"
aaa=[]
for i in dataset["position"]:
    aaa.append(("Data Scientist" in i))
dataset["Data Scientist"]=aaa
dataset.head()

position  ... Data Scientist
0                               Development Director  ...          False
1  An Ostentatiously-Excitable Principal Research...  ...          False
2                                     Data Scientist  ...           True
3                                       Data Analyst  ...          False
4  Assistant Professor -TT - Signal Processing & ...  ...          False

[5 rows x 8 columns]

In [0]:
dataset["Data Scientist"].value_counts()

False    5710
True     1243
Name: Data Scientist, dtype: int64

In [0]:
dataset[(dataset['company']=='Amazon.com') & (dataset['Data Scientist']==True)]

position  ... Data Scientist
1329       Data Scientist - Natural Language Processing  ...           True
1336                                     Data Scientist  ...           True
1343                                 NLP Data Scientist  ...           True
2503                            Research Data Scientist  ...           True
2515                  Research Data Scientist- Alexa ML  ...           True
3555                                  Sr Data Scientist  ...           True
4388                                     Data Scientist  ...           True
4978         Data Scientist, Fire TV Product Management  ...           True
4979                 Data Scientist, Alexa Shopping NLU  ...           True
4980              Data Scientist, Private Label Apparel  ...           True
5009                            Data Scientist - Delphi  ...           True
5012                                     Data Scientist  ...           True
5022                           Data Scientist, FBA fees  ...           True
5023                       Data Scientist - Risk Mining  ...           True
5025               Data Scientist II - Payment Products  ...           True
5041  Data Scientist, Natural Language Processing (N...  ...           True
5043     Data Scientist - Amazon Business International  ...           True
5045  Sr. Data Scientist - Data Visualization Specia...  ...           True
5051                              Senior Data Scientist  ...           True
5052  Data Scientist- Device Analytics Research Econ...  ...           True
5069                    Senior Data Scientist, Alexa AI  ...           True
5092              Sr. Data Scientist - Machine Learning  ...           True
5096  Principal Data Scientist - Prime Video Persona...  ...           True
6704                              Senior Data Scientist  ...           True
6936       Data Scientist - Alexa Language Technologies  ...           True
6939  Data Scientist, Customer Interests & Personali...  ...           True

[26 rows x 8 columns]

***Note - ***


1.   "Amazon.com" company maximum goes at location "Seattle, WA" for offering "Data Scientist" job. 



***Analysis upto now - ***

1.   Maximum job offers are of "Data Scientist" position. 
2.   "Amazon.com" company is providing maximum job oppurtunities.  
3.   "Amazon.com" company maximum goes at location "Seattle, WA" for offering "Data Scientist" job.
4.   Maximum offers are at the location "Seattle, WA". But the company with maximum reviews go at "Sunnyvale,CA 94086" location instead of going at location "Seattle, WA". This means that more talented persons are at "Sunnyvale, CA 94086" location. 
5.   "New York" city has maximum job offers. And "Seattle" city is offering second heighest job offers. 
6.   "CA" state has maximum job offers.


# ***Encoding Categorical Data - ***

In [0]:
dataset.shape

(6953, 8)

In [0]:
X=dataset
print(X.head())
print(X.shape)

                                            position  ... Data Scientist
0                               Development Director  ...          False
1  An Ostentatiously-Excitable Principal Research...  ...          False
2                                     Data Scientist  ...           True
3                                       Data Analyst  ...          False
4  Assistant Professor -TT - Signal Processing & ...  ...          False

[5 rows x 8 columns]
(6953, 8)


In [0]:
X.columns

Index(['position', 'company', 'description', 'reviews', 'location', 'City',
       'State', 'Data Scientist'],
      dtype='object')

In [0]:
X=X.drop(labels=['position','description','City','State','Data Scientist'],axis=1)
X.columns

Index(['company', 'reviews', 'location'], dtype='object')

In [0]:
print(X.head())

              company  reviews            location
0             ALS TDI        0  Atlanta, GA 30301 
1  The Hexagon Lavish        0         Atlanta, GA
2      Xpert Staffing        0         Atlanta, GA
3      Operation HOPE       44  Atlanta, GA 30303 
4    Emory University      550         Atlanta, GA


In [0]:
Y=dataset
print(Y.head())
print(Y.shape)

                                            position  ... Data Scientist
0                               Development Director  ...          False
1  An Ostentatiously-Excitable Principal Research...  ...          False
2                                     Data Scientist  ...           True
3                                       Data Analyst  ...          False
4  Assistant Professor -TT - Signal Processing & ...  ...          False

[5 rows x 8 columns]
(6953, 8)


In [0]:
print(Y.columns)

Index(['position', 'company', 'description', 'reviews', 'location', 'City',
       'State', 'Data Scientist'],
      dtype='object')


In [0]:
Y=Y.drop(labels=['position', 'company', 'description', 'reviews', 'location', 'City','State'],axis=1)
print(Y.head())
print(Y.shape)
print(Y.columns)

   Data Scientist
0           False
1           False
2            True
3           False
4           False
(6953, 1)
Index(['Data Scientist'], dtype='object')


In [0]:
print(X.columns)
print(Y.columns)

Index(['company', 'reviews', 'location'], dtype='object')
Index(['Data Scientist'], dtype='object')


In [0]:
# Applying labelencoder on X
labelencoder_X=LabelEncoder() 
X.iloc[:,0]=labelencoder_X.fit_transform(X.iloc[:,0])
X.iloc[:,2]=labelencoder_X.fit_transform(X.iloc[:,2])
print(X)

      company  reviews  location
0          34        0         4
1        1924        0         3
2        2153        0         3
3        1443       44         5
4         687      550         3
5        1248      385        10
6        1248      385        10
7        2137      899         3
8        1838     3343         3
9        1095     4494        18
10       1925    38844         3
11        429      390         3
12       1095     4494        18
13        687      550         3
14       1377      912         3
15        522      104         3
16        330      152         8
17       1501       44        18
18       1625        0         3
19       1028        0         3
20        330      152         8
21       1095     4494        18
22       1396      722         8
23       1925    38844         3
24        713      447         3
25         88        0         3
26       1925        0         3
27        847      496        10
28       1925    38844         3
29        

In [0]:
# Applying OneHotEncoder on X
onehotencoder=OneHotEncoder(categorical_features=[0])
X=onehotencoder.fit_transform(X).toarray()
onehotencoder=OneHotEncoder(categorical_features=[2])
X=onehotencoder.fit_transform(X).toarray()

/usr/local/lib/python3.6/dist-packages/sklearn/preprocessing/_encoders.py:415: FutureWarning: The handling of integer data will change in version 0.22. Currently, the categories are determined based on the range [0, max(values)], while in the future they will be determined based on the unique values.
If you want the future behaviour and silence this warning, you can specify "categories='auto'".
In case you used a LabelEncoder before this OneHotEncoder to convert the categories to integers, then you can now use the OneHotEncoder directly.
  warnings.warn(msg, FutureWarning)
/usr/local/lib/python3.6/dist-packages/sklearn/preprocessing/_encoders.py:451: DeprecationWarning: The 'categorical_features' keyword is deprecated in version 0.20 and will be removed in 0.22. You can use the ColumnTransformer instead.
  "use the ColumnTransformer instead.", DeprecationWarning)
/usr/local/lib/python3.6/dist-packages/sklearn/preprocessing/_encoders.py:415: FutureWarning: The handling of integer data w

In [0]:
print(X)

[[1.000e+00 0.000e+00 0.000e+00 ... 0.000e+00 0.000e+00 4.000e+00]
 [1.000e+00 0.000e+00 0.000e+00 ... 0.000e+00 0.000e+00 3.000e+00]
 [1.000e+00 0.000e+00 0.000e+00 ... 0.000e+00 0.000e+00 3.000e+00]
 ...
 [1.000e+00 0.000e+00 0.000e+00 ... 0.000e+00 5.000e+00 3.430e+02]
 [1.000e+00 0.000e+00 0.000e+00 ... 0.000e+00 4.618e+03 3.430e+02]
 [1.000e+00 0.000e+00 0.000e+00 ... 0.000e+00 1.161e+04 3.460e+02]]


In [0]:
X=X.astype(int)
print(X)

[[    1     0     0 ...     0     0     4]
 [    1     0     0 ...     0     0     3]
 [    1     0     0 ...     0     0     3]
 ...
 [    1     0     0 ...     0     5   343]
 [    1     0     0 ...     0  4618   343]
 [    1     0     0 ...     0 11610   346]]


In [0]:
# Applying labelencoder on Y 
labelencoder_Y=LabelEncoder()
Y.iloc[:,0]=labelencoder_Y.fit_transform(Y.iloc[:,0])
print(Y)

      Data Scientist
0                  0
1                  0
2                  1
3                  0
4                  0
5                  0
6                  0
7                  0
8                  0
9                  1
10                 0
11                 0
12                 1
13                 0
14                 0
15                 1
16                 0
17                 0
18                 1
19                 1
20                 0
21                 0
22                 1
23                 0
24                 1
25                 1
26                 1
27                 1
28                 1
29                 1
...              ...
6934               1
6935               0
6936               1
6937               1
6938               1
6939               1
6940               1
6941               1
6942               0
6943               1
6944               1
6945               0
6946               0
6947               0
6948               0
6949         

# ***Splitting data into train and test data - ***

In [0]:
X_train,X_test,Y_train,Y_test=train_test_split(X,Y,test_size=0.25,random_state=0)
print(X_train,X_test,Y_train,Y_test)

[[   1    0    0 ...    0  530   79]
 [   1    0    0 ...    0 6318  231]
 [   1    0    0 ...    0    4  290]
 ...
 [   1    0    0 ...    0  211  110]
 [   1    0    0 ...    0   15   79]
 [   1    0    0 ...    0  596   79]] [[   1    0    0 ...    0    5  343]
 [   1    0    0 ...    0 4214  290]
 [   1    0    0 ...    0 2473  317]
 ...
 [   1    0    0 ...    0    0   52]
 [   1    0    0 ...    0    0  135]
 [   1    0    0 ...    0    0   52]]       Data Scientist
2612               0
4323               1
5948               0
2276               0
4175               1
5340               0
2071               0
3992               0
5815               1
4431               0
3955               0
5732               0
6416               0
4636               0
5084               0
6599               0
4223               0
159                0
5462               0
2233               1
3417               0
6108               0
192                0
2847               0
1910               

# ***Applying Logistic Regression - ***

In [0]:
classifier=LogisticRegression()
classifier.fit(X_train,Y_train)

/usr/local/lib/python3.6/dist-packages/sklearn/linear_model/logistic.py:432: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
/usr/local/lib/python3.6/dist-packages/sklearn/utils/validation.py:724: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().
  y = column_or_1d(y, warn=True)


LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=100,
                   multi_class='warn', n_jobs=None, penalty='l2',
                   random_state=None, solver='warn', tol=0.0001, verbose=0,
                   warm_start=False)

In [0]:
# Making Predictions 
Y_pred=classifier.predict(X_test)

# ***Measuring Performance - ***

***1. Accuracy Score - ***

In [0]:
metrics.accuracy_score(Y_test,Y_pred)

0.8096607245543416

***2. Classification Report - ***

In [0]:
print(metrics.classification_report(Y_test,Y_pred))

              precision    recall  f1-score   support

           0       0.81      1.00      0.89      1406
           1       0.67      0.01      0.02       333

    accuracy                           0.81      1739
   macro avg       0.74      0.51      0.46      1739
weighted avg       0.78      0.81      0.73      1739

